floors/enforce.go

```diff
@@ -0,0 +1,45 @@
+package floors
+
+import (
+	"github.com/mxmCherry/openrtb/v16/openrtb2"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+func ShouldEnforce(bidRequest *openrtb2.BidRequest, floorExt *openrtb_ext.PriceFloorRules, configEnforceRate int, f func(int) int) bool {
+
+	if floorExt != nil && floorExt.Skipped != nil && *floorExt.Skipped {
+		var floorInRequest bool
+		for i := range bidRequest.Imp {
+			if bidRequest.Imp[i].BidFloor > 0 {
+				floorInRequest = true
+				break
+			}
+		}
+		if !floorInRequest {
+			return floorInRequest
+		}
+	}
+
+	if floorExt != nil && floorExt.Enforcement != nil && floorExt.Enforcement.EnforcePBS != nil && !*floorExt.Enforcement.EnforcePBS {
+		return *floorExt.Enforcement.EnforcePBS
+	}
+
+	if floorExt != nil && floorExt.Enforcement != nil && floorExt.Enforcement.EnforceRate > 0 {
+		configEnforceRate = floorExt.Enforcement.EnforceRate
+	}
+
+	shouldEnforce := configEnforceRate > f(ENFORCE_RATE_MAX+1)
+	if floorExt == nil {
+		floorExt = new(openrtb_ext.PriceFloorRules)
+	}
+
+	if floorExt.Enforcement == nil {
+		floorExt.Enforcement = new(openrtb_ext.PriceFloorEnforcement)
+	}
+
+	if floorExt.Enforcement.EnforcePBS == nil {
+		floorExt.Enforcement.EnforcePBS = new(bool)
+	}
+	*floorExt.Enforcement.EnforcePBS = shouldEnforce
```

Comment:
Why does this update the floorExt in addition to returning the value? Can we just use the return value and not modify the request?

Reply:
This update is done to handle below case:
When Enforcement.EnforcePBS  is not defined and based on EnforceRate enforcement needs to be done or skipped, 
In that case this value is updated with appropriate value. 
And to update prebidExt due to this case requestExt.SetPrebid(prebidExt) is done in enforceFloors()

Reply:
No other code in this PR check for `EnforcePBS`. Is this therefore a signal downstream to bidders so they know what behavior PBS performed? @bretg checking on the spec intention here.